### ISSUE TITLE
[Bug] The app crashes when switching themes

### ISSUE THREAD

**Issue by Amejia481** [mozilla-mobile/fenix#2449]
## Steps to reproduce
1. Go to Settings.
1. Theme.
1. Select a new theme.

### Expected behavior
The theme is switched.

### Actual behavior
The app crashes.

### Stack trace
```
 kotlin.UninitializedPropertyAccessException: lateinit property sessionControlComponent has not been initialized
        at org.mozilla.fenix.home.HomeFragment.onSaveInstanceState(HomeFragment.kt:170)
        at androidx.fragment.app.Fragment.performSaveInstanceState(Fragment.java:2799)
        at androidx.fragment.app.FragmentManagerImpl.saveFragmentBasicState(FragmentManagerImpl.java:2276)
        at androidx.fragment.app.FragmentManagerImpl.saveAllState(FragmentManagerImpl.java:2335)
        at androidx.fragment.app.Fragment.performSaveInstanceState(Fragment.java:2802)
        at androidx.fragment.app.FragmentManagerImpl.saveFragmentBasicState(FragmentManagerImpl.java:2276)
        at androidx.fragment.app.FragmentManagerImpl.saveAllState(FragmentManagerImpl.java:2335)
        at androidx.fragment.app.FragmentController.saveAllState(FragmentController.java:151)
        at androidx.fragment.app.FragmentActivity.onSaveInstanceState(FragmentActivity.java:513)
        at androidx.appcompat.app.AppCompatActivity.onSaveInstanceState(AppCompatActivity.java:533)
        at android.app.Activity.performSaveInstanceState(Activity.java:1550)
        at android.app.Instrumentation.callActivityOnSaveInstanceState(Instrumentation.java:1443)
        at android.app.ActivityThread.callActivityOnSaveInstanceState(ActivityThread.java:4848)
        at android.app.ActivityThread.callActivityOnStop(ActivityThread.java:4195)
        at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:4796)
        at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4732)
        at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:69)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ClientTransactionHandler.executeTransaction(ClientTransactionHandler.java:55)
        at android.app.ActivityThread.handleRelaunchActivityLocally(ActivityThread.java:4782)
        at android.app.ActivityThread.access$3200(ActivityThread.java:200)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1826)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6718)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

**Comment by colintheshots** [comment]
This seems to be because onSaveInstanceState is getting called before onCreateView. I'll fix this.

**Comment by ekager** [comment]
This was fixed b #2450 , closing
